### PR TITLE
feat: :sparkles: add `drop_property_fields()`

### LINKED ISSUE #622: Create and test `drop_property_fields()` - a part of `create_properties_template()`
Signature: `drop_property_fields(properties: json, field: List[str]) -> json:`

Arguments:
- `properties`: json with properties
- `field`: List of strings with fields to drop

Returns: json with properties without the fields that has been dropped

Context: see the `create_properties_template` [diagram](https://github.com/seedcase-project/seedcase-sprout/blob/main/docs/design/images/core-python-functions/create-properties-template.svg)

### PR DESCRIPTION
**Pull request by martonvago**
## Description

This adds `drop_property_fields()`.

Closes #622 

## Reviewer Focus

This PR needs a quick review.

## Checklist

- [x] Added or updated tests
- [x] Tests passed locally
- [x] Linted and formatted code
- [x] Build passed locally
- [x] Updated documentation

### CODE REVIEW

File: sprout/core/drop_property_fields.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+def drop_property_fields(properties: dict, fields: list[str]) -> dict:
+    """Deletes the specified fields from the properties object.
+
+    Ignores fields not in properties.
```
Comment (martonvago) [inline sprout/core/drop_property_fields.py:4]: Seemed sensible to do it this way

**Review by signekb — COMMENTED** [review]
Very nice 🎉 
I have the smallest comment ever. But apart from that, I approve ✅

File: tests/core/test_drop_property_fields.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def test_drops_multiple_fields():
+    """Given multiple existing fields to drop, should return the input
+    without those fields."""
+    properties = {
+        "test1": "value1",
+        "test2": "value2",
+        "test3": "value3",
+        "test4": "value4",
```
[the remark this answers is not in the document]
Comment (martonvago) [reply tests/core/test_drop_property_fields.py]: Sure, no particular reason :)

**Review by lwjohnst86 — APPROVED** [review]
Very nice! I started to write a comment about the for loop, but did some reading and it seems, at least for modifying dict objects, this is the fastest and cleanest way to do it. So Awesome job 🌠

**Pull request merged**